### DISCUSSION TITLE
Branch Protections for new branches

### DISCUSSION THREAD

**Discussion by ghost** [community/community#12311] [Pull Requests]
It was recommended that we should use use a "branching", not
"forking", workflow for our internal/private GitHub repositories. This
means that: all developers are given write access to all repositories,
and any important branch is protected using branch protections. This
allows developers to open pull-requests from non-protected
branches. The issue is that developers are able to create any branch
that they choose even in a "protected namespace". For example a
developer could create a `release/2.0.0` branch that is protected
under the `release/*` branch protection rule. After creation the
developer is unable to update that branch since it is protected and
administrators cannot delete that branch since it is protected.

I have 2 questions:
 - How can we go about restricting branch creation?
 - Can we apply the same restrictions to branch creation as our branch
   protections do?

**Comment by bero** [comment]
I have also the same problem. Here are docs how to create branch protection https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule

But I cannot see any setting for branches. I assume some permission is missing
